src/main/java/runge/wordledictionary/WordleGameFrame.java

```diff
@@ -2,46 +2,132 @@
+        JLabel wordleLabel = new JLabel(image);
+        wordleLabel.setHorizontalAlignment(JLabel.CENTER);
+        mainPanel.add(wordleLabel, BorderLayout.NORTH);
+
+
+        //grid
+        JPanel gridPanel = new JPanel();
+        gridPanel.setLayout(new GridLayout(6, 5, 20, 20)); //add two more ints for gaps
+        for (int i = 0; i < 6; i++)
+        {
+            for (int j = 0; j < 5; j++)
+            {
+                JLabel letterBox = new JLabel();
+                letterBox.setBorder(BorderFactory.createLineBorder(Color.black, 1));
+                letters[i][j] = letterBox;
+                gridPanel.add(letters[i][j]);
+            }
+        }
+        mainPanel.add(gridPanel, BorderLayout.CENTER);
+
+
+        //keyboard
+        JPanel keyboardGrid = new JPanel(new GridLayout(3, 10));
+        JPanel keyboardLine1 = new JPanel();
+        JPanel keyboardLine2 = new JPanel();
+        JPanel keyboardLine3 = new JPanel();
+        enter = new JButton("Return");
+        backspace = new JButton("Delete");
+        String[] row1 = {"Q", "W", "E", "R", "T", "Y", "U", "I", "O", "P"};
+        String[] row2 = {"A", "S", "D", "F", "G", "H", "J", "K", "L"};
+        String[] row3 = {"Z", "X", "C", "V", "B", "N", "M"};
+        StringBuilder addRows = new StringBuilder();
+
+        for (int i = 0; i < row1.length; i++)
+        {
+            keyboard[i] = new JButton(row1[i]);
+            keyboardLine1.add(keyboard[i]);
+        }
+        for (int j = 0; j < row2.length; j++)
+        {
+            keyboard[j] = new JButton(row2[j]);
+            keyboardLine2.add(keyboard[j]);
+        }
+        for (int k = 0; k < row3.length; k++)
+        {
+            keyboard[k] = new JButton(row3[k]);
+            keyboardLine3.add(keyboard[k]);
+        }
+
+
+        keyboardLine2.add(enter);
+        keyboardLine3.add(backspace);
+        keyboardGrid.add(keyboardLine1, BorderLayout.NORTH);
+        keyboardGrid.add(keyboardLine2, BorderLayout.CENTER);
+        keyboardGrid.add(keyboardLine3, BorderLayout.SOUTH);
+        mainPanel.add(keyboardGrid, BorderLayout.SOUTH);
+
+        setFocusable(true);
+        requestFocus();
 
-        //label that shows output of word guessed
-        JLabel guessLabel = new JLabel();
-        mainPanel.add(guessLabel, BorderLayout.CENTER);
+        //virtual keyboard
+        //need to go through each button and add action listener
 
-        //button that once pressed sends guess field into guess method then outputs to label
-        JButton button = new JButton("Guess");
-        mainPanel.add(button, BorderLayout.SOUTH);
+        for (int i = 0; i < row1.length; i++)
+        {
+            //only adding last line... z,x,c...
+            keyboard[i].addActionListener(e ->
+                    System.out.println("Letter typed " + e.getActionCommand()));
+        }
```

Comment:
How many elements are actually in the keyboard array?

Hint: look at it with the debugger.

Reply:
it says there are 9 things in the array. but i thought that once i added more, it should fill the array?

Reply:
Maybe you're not filling the array like you think you are.

Reply:
that's what I figured, but I can't seem to figure it out. I tried changing the way I fill it a few different times but either I get all my variables on one line or they just repeat the letter Q

Reply:
I figured it out